runtime/compositeschema_test.go

```diff
@@ -91,5 +88,112 @@ func TestInvalidSchemaReference(t *testing.T) {
 	if err == nil {
 		t.Fatal("Error not returned indicating a composite schema could not be created")
 	}
+}
+
+func TestCreateMergedCompositeSchemas(t *testing.T) {
+	t.Parallel()
+
+	c1, err := NewCompositeSchema("prop1", `{
+		"type": "object",
+        "properties": {
+			"count": {"type": "integer"}
+        },
+		"additionalProperties": false
+	}`, true, func() interface{} { return nil })
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c2, err := NewCompositeSchema("prop2", `{
+		"type": "object",
+        "properties": {
+			"count": {"type": "integer"}
+        },
+		"additionalProperties": false
+	}`, true, func() interface{} { return nil })
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err = MergeCompositeSchemas(c1, c2); err != nil {
+		t.Fatalf("Error creating a merged composite scheme. %v", err)
+	}
+}
+
+func TestMergeComposedSchemas(t *testing.T) {
+	t.Parallel()
+
+	schema := `
+		{
+			"type": "object",
+			"properties": {
+				"count": {"type": "integer"}
+			},
+			"additionalProperties": false
+		}
+	`
+
+	composites := make([]CompositeSchema, 4)
+	for i := range composites {
+		// Create unique property names to avoid collision resulting in error
+		propertyName := fmt.Sprintf("prop%d", i)
+		composite, err := NewCompositeSchema(propertyName, schema, true, func() interface{} { return nil })
+		if err != nil {
+			t.Fatal(err)
+		}
+		composites[i] = composite
+	}
+
+	composed1, err := MergeCompositeSchemas(composites[0:2]...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	composed2, err := MergeCompositeSchemas(composites[2:]...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	merged, err := MergeCompositeSchemas(composed1, composed2)
+	if err != nil {
+		t.Fatalf("Error creating a merged composite scheme. %v", err)
+	}
 
+	m := merged.(*composedSchema)
+	if len(m.entries) != 2 {
+		t.Fatalf("Merged schema should have 4 entries but had %d", len(m.entries))
```

Comment:
nope... does this test pass? You are merging in a tree... So you should only have two... as both of those each has two...